aws-lc-fips-sys/builder/cmake_builder.rs

```diff
@@ -174,8 +174,31 @@ impl CmakeBuilder {
                     "CMAKE_TOOLCHAIN_FILE",
                     format!("{ndk}/native/build/cmake/ohos.toolchain.cmake"),
                 );
-                cmake_cfg.define("CMAKE_C_FLAGS", "-Wno-unused-command-line-argument");
-                cmake_cfg.define("CMAKE_CXX_FLAGS", "-Wno-unused-command-line-argument");
+                match target().as_str() {
+                    "aarch64-unknown-linux-ohos" => {
+                        cmake_cfg
+                            .cflag("-Wno-unused-command-line-argument")
+                            .cxxflag("-Wno-unused-command-line-argument");
+                    }
+                    "armv7-unknown-linux-ohos" => {
+                        cmake_cfg
+                        .cflag("-Wno-unused-command-line-argument -march=armv7-a -mfloat-abi=softfp -mtune=generic-armv7-a -mthumb -mfpu=neon -DHAVE_NEON")
+                        .cxxflag("-Wno-unused-command-line-argument -march=armv7-a -mfloat-abi=softfp -mtune=generic-armv7-a -mthumb -mfpu=neon -DHAVE_NEON")
+                        .asmflag("-march=armv7-a -mfloat-abi=softfp -mtune=generic-armv7-a -mthumb -mfpu=neon -DHAVE_NEON");
+                    }
+                    "x86_64-unknown-linux-ohos" => {
+                        cmake_cfg
+                            .cflag("-Wno-unused-command-line-argument -msse4.1 -DHAVE_NEON_X86 -DHAVE_NEON")
+                            .cxxflag("-Wno-unused-command-line-argument -msse4.1 -DHAVE_NEON_X86 -DHAVE_NEON")
+                            .asmflag("-msse4.1 -DHAVE_NEON_X86 -DHAVE_NEON");
+                        // we must set those env for cross-build in apple silicon machine
+                        cmake_cfg.define("CMAKE_SYSTEM_PROCESSOR", "x86_64");
+                        cmake_cfg.define("CMAKE_OSX_ARCHITECTURES", "x86_64");
+                    }
```

Comment:
* I'm concerned that these compiler flags are specific to the hardware that you are building for and not necessarily appropriate for all hardware relevant to the given target.
* Could these flags instead be specified in the build environment using `CMAKE_C_FLAGS` and `CMAKE_CXX_FLAGS`.  (I think our build setup was wrong to set "define" these values w/o regard to the value provided by the environment.)

Reply:
1. The parameters for the armv7a architecture are in line with the official Rust build parameters. https://doc.rust-lang.org/rustc/platform-support/openharmony.html
2. When it comes to the x86_64 architecture, I believe we've reached the most streamlined configuration. For information on SSE version compatibility, the official OpenHarmony document is your go-to guide. https://docs.openharmony.cn/pages/v4.1/en/application-dev/napi/neon-guide.md
3. I've chosen not to use CMAKE_C_FLAGS because it would lead to a failed build for the armv7a architecture. Do you have any smart fixes up your sleeve?
<img width="1292" alt="image" src="https://github.com/aws/aws-lc-rs/assets/32590310/cfd5f583-6722-4f7b-a9b9-91fb300778fc">